packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelMain.tsx

```diff
@@ -378,6 +411,20 @@ export function TransferPanelMain({
     setQueryParams({ l2ChainId })
   }, [isConnectedToArbitrum, externalFrom, externalTo, setQueryParams])
 
+  useEffect(() => {
+    // If a different destination address is specified, we want to update the ERC-20 balance accordingly
+    if (selectedToken && isAddress(destinationAddressOrWalletAddress)) {
+      updateErc20L1Balance([selectedToken.address])
+      updateErc20L2Balance([String(selectedToken.l2Address)])
```

Comment:
I don't think we should cast here, if `selectedToken.l2Address` is undefined, we don't want to update balance for `undefined` address

Reply:
this is related to #812 
I think in the case for tokens without an L2 address, the function should return something specific and indicate it's unbridged on the UI
but for this PR i think we can cast it for now
deal with it on a PR for #812